src/TraceEvent/TraceLog.cs

```diff
@@ -5800,6 +5800,15 @@ internal int ParentDepth()
 #endif
         #endregion
 
+        internal bool IsKernelAddress(Address ip, int pointerSize)
+        {
+            // EventPipe doesn't generate kernel address events and current heauristics are not deterministic on none Windows platforms.
+            if (log?.rawEventSourceToConvert is EventPipeEventSource)
```

Comment:
Overall this strategy is a fine way to do this.  I would like to not put this method on `TraceProcess` since it isn't really a per-process concept - it is a per-trace concept.  Can you move this method to `TraceLog`?  It can still be accessed from `process.Log`.  You will probably need to rename it, perhaps to something like `IsWindowsKernelAddress`.

Reply:
@brianrob Renamed TraceLog.IsKernelAddress to TraceLog.IsWindowsKernelAddress.